azkaban-common/src/main/java/azkaban/project/ProjectManager.java

```diff
@@ -360,6 +360,7 @@ public void setJobOverrideProperty(Project project, Props prop, String jobName,
         projectLoader.fetchProjectProperty(project, prop.getSource());
 
     if (oldProps == null) {
+      oldProps = new Props();
       projectLoader.uploadProjectProperty(project, prop);
```

Comment:
@wangqiaoshi the if else looks redundant. Is it possible that we just move projectLoader.uploadProjectProperty(project, prop) outside of if else logics ?